src/gui/BandSplitter/TriStateButtonAttachment.cpp

```diff
@@ -6,9 +6,16 @@ namespace gui::band_splitter
 {
 TriStateButtonAttachment::TriStateButtonAttachment (chowdsp::BoolParameter& threeBandParam,
                                                     chowdsp::BoolParameter& fourBandParam,
+                                                    chowdsp::PluginState& pluginState,
```

Comment:
I think we could get the `threeBandParam` and `fourBandParam` from the `pluginState` (i.e. `pluginState.params.bandSplitterParams.threeBandParam`, or something like that). It's also 100% fine to pass them explicitly as arguments... whichever feels better to you.

Reply:
Ah I see, the reason I couldn't do this currently is I had the type as` chowdsp::PluginState&` and not `State&`, I've changed it now and can avoid passing the BoolParameters to `TriStateButtonAttatchment` and `TriStateButton` :)